File: app/src/main/java/com/esafirm/androidplayground/securities/SecurityMenuController.java
package com.esafirm.androidplayground.securities;

import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bluelinelabs.conductor.RouterTransaction;
import com.esafirm.androidplayground.common.BaseController;
import com.esafirm.androidplayground.common.MenuFactory;
import com.esafirm.androidplayground.common.OnNavigatePage;

import java.util.Arrays;

public class SecurityMenuController extends BaseController {

    @NonNull
    @Override
    protected View onCreateView(@NonNull LayoutInflater inflater, @NonNull ViewGroup container) {
        return MenuFactory.create(container.getContext(), Arrays.asList(
                "Bcrypt"
        ), new OnNavigatePage() {
            @Override
            public void navigate(int index) {
                getRouter().pushController(RouterTransaction.with(new BcryptController()));
            }
        });
    }
}

File: app/src/main/java/com/esafirm/androidplayground/securities/SecurityMenuController.kt
package com.esafirm.androidplayground.securities

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bluelinelabs.conductor.RouterTransaction
import com.esafirm.androidplayground.common.BaseController
import com.esafirm.androidplayground.common.MenuFactory

class SecurityMenuController : BaseController() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup): View {
        return MenuFactory.create(
            container.context, listOf(
                "Bcrypt",
                "Facebook Conceal",
                "AES Encrypt Decrypt",
            )
        ) {
            router.pushController(
                RouterTransaction.with(
                    when (it) {
                        0 -> BcryptController()
                        1 -> ConcealController()
                        2 -> AesEncryptDecryptController()
                        else -> error("No controller for $it")
                    }
                )
            )
        }
    }
}
